include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -674,14 +694,18 @@ struct __parallel_reduce_then_scan_scan_submitter<
                     if (__block_num + 1 == __num_blocks)
                     {
                         __res_ptr[0] = __sub_group_carry.__v;
+
+                        if constexpr (__is_unique_pattern)
```

Comment:
Nitpick here but maybe it's better to make an `if constexpr ... else` conditional so we aren't writing to the same global pointer twice.

Maybe something like:
```
if constexpr (__is_unique_pattern)
{
    __res_ptr[0] = __sub_group_carry.__v + 1;
}
else
{
     __res_ptr[0] = __sub_group_carry.__v;
}
```

Reply:
fixed.